packages/electron/src/electron-main/electron-window-service.ts

```diff
@@ -0,0 +1,36 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { shell } from 'electron';
+import { injectable, inject } from 'inversify';
+import { ElectronMainWindowService } from '../electron-common/electron-window-protocol';
+import { ElectronApplication } from './electron-application';
+
+@injectable()
+export class DefaultElectronMainWindowService implements ElectronMainWindowService {
+
+    @inject(ElectronApplication)
+    protected readonly app: ElectronApplication;
+
+    openElectronWindow(url: string): void {
+        this.app.openWindowWithWorkspace(url);
```

Comment:
I think the goal of the `openElectronWindow` is to actually open an electron window with some URL. Here it delegates the call to `app.openWindowWithWorkspace(url)` which assumes that `url` is the fs path to a workspace, not the full URL required for the window to open. It might be a mistake I made?

Reply:
> It might be a mistake I made?

Your code did not work. Please check on your branch and let's adjust it if you want this feature. You can easily reproduce by opening a workspace then another. The second URL was something like this: `http://localhost#some/path/to/workspace/#then/the/second/I/have/opened` which did not work.

Reply:
Or I am overlooking something 😄 there was so many changes already.